src/auto_archiver/archivers/base_archiver/base_archiver.py

```diff
@@ -0,0 +1,291 @@
+        result.set_title(video_data.pop('title', video_data.pop('fulltitle', "")))
+
+        # then add the platform specific additional metadata
+        for key, mapping in self.video_data_metadata_mapping(extractor_key, video_data).items():
+            if isinstance(mapping, str):
+                result.set(key, eval(f"video_data{mapping}"))
+            elif callable(mapping):
+                result.set(key, mapping(video_data))
+        result.set_url(url)
+
+        # extract comments if enabled
+        if self.comments:
+            result.set("comments", [{
+                "text": c["text"],
+                "author": c["author"], 
+                "timestamp": datetime.datetime.fromtimestamp(c.get("timestamp"), tz = datetime.timezone.utc)
+            } for c in video_data.get("comments", [])])
+
+        # then add the common metadata
+        if timestamp := video_data.pop("timestamp", None):
+            timestamp = datetime.datetime.fromtimestamp(timestamp, tz = datetime.timezone.utc).isoformat()
+            result.set_timestamp(timestamp)
+        if upload_date := video_data.pop("upload_date", None):
+            upload_date = datetime.datetime.strptime(upload_date, '%Y%m%d').replace(tzinfo=datetime.timezone.utc)
+            result.set("upload_date", upload_date)
+        
+        # then clean away any keys we don't want
+        for clean_key in self.keys_to_clean(extractor_key, video_data):
+            video_data.pop(clean_key, None)
+        
+        # then add the rest of the video data
+        for k, v in video_data.items():
+            if v:
+                result.set(k, v)
+
+        return result
+
+    def video_data_metadata_mapping(self, extractor_key: str, video_data: dict) -> dict:
+        """
+        Returns a key->value mapping to map from the yt-dlp produced 'video_data' to the Metadata object.
+        Can be either a string for direct mapping, or a function, or a lambda.
+        """
+        return {}
+    
+    def suitable_extractors(self, url: str) -> list[str]:
+        """
+        Returns a list of valid extractors for the given URL"""
+        for info_extractor in yt_dlp.YoutubeDL()._ies.values():
+            if info_extractor.suitable(url) and info_extractor.working():
+                yield info_extractor
+        
+    def suitable(self, url: str) -> bool:
+        """
+        Checks for valid URLs out of all ytdlp extractors.
+        Returns False for the GenericIE, which as labelled by yt-dlp: 'Generic downloader that works on some sites'
+        """
+        return any(self.suitable_extractors(url))
+
+    def create_metadata_for_post(self, info_extractor: InfoExtractor, video_data: dict, url: str) -> Metadata:
+        """
+        Standardizes the output of the ytdlp InfoExtractor to a common format
+        """
+        if info_extractor.ie_key() == 'Bluesky':
+            return bluesky.create_metadata(video_data, self, url)
```

Comment:
am I correct in assuming that when getting the metadata from the info_extractor for post (vs for video) that the unnecessary keys cleaned up above are not present, or any form of those?

Reply:
You are correct, the unecessary keys are only for the 'video_data' which is the format returned by ytdlp for valid videos.

I don't think there's any harm in trying to remove the keys that don't exist - a few extra CPU cycles. But the next step would be to extend the `keys_to_clean` method to remove keys from the bluesky `post` json (i.e. the API response) that we also don't want. I realise I haven't done that here.

Reply:
Actually this is what the bsky `post` API gets. They look fine - nothing to clean. Below is bsky and twitter - let me know if you think anything else needs removing from either of them?
<img width="1308" alt="Screen Region 2025-01-17 at 11 49 15" src="https://github.com/user-attachments/assets/d55a2271-65de-4d33-9f86-d5c88d43d7de" />
<img width="1315" alt="Screen Region 2025-01-17 at 11 51 42" src="https://github.com/user-attachments/assets/ed7b935f-9a45-48c6-95d2-1257e444ad22" />